tests/Fixer/Operator/NoUselessConcatOperatorFixerTest.php

```diff
@@ -0,0 +1,176 @@
+            'double . single' => ['"b\n"', "'a'", '"b\na"'],
+            'double . double' => ['"b"', '"b"', '"bb"'],
+            'encapsed . encapsed' => ['"{$a}"', '"{$b}"', '"{$a}{$b}"'],
+            'encapsed space. encapsed' => ['"{$a1} "', '"{$b2}"', '"{$a1} {$b2}"'],
+            'encapsed . space encapsed' => ['"{$a}"', '" {$b}"', '"{$a} {$b}"'],
+            'encapsed space. space encapsed' => ['"{$a} "', '" {$b}"', '"{$a}  {$b}"'],
+            'encapsed . single' => ['"{$a}"', "'Z'", '"{$a}Z"'],
+            'single . encapsed' => ["'Y'", '"{$a}"', '"Y{$a}"'],
+            'encapsed spaced. single' => ['"{$a}   "', "'G'", '"{$a}   G"'],
+            'single . space encapsed' => ["'V'", '" {$a}"', '"V {$a}"'],
+            'encapsed . double' => ['"{$a} "', '"XX"', '"{$a} XX"'],
+            'double . encapsed' => ['"XCV"', '"{$a}"', '"XCV{$a}"'],
+            'encapsed spaced . double' => ['"{$a} V "', '"PO"', '"{$a} V PO"'],
+            'double . space encapsed' => ['"DSA"', '"   XX {$a}"', '"DSA   XX {$a}"'],
+        ];
+
+        foreach ($cases as $label => $case) {
+            yield $label => [
+                sprintf($templateExpected, $case[2]),
+                sprintf($templateInput, $case[0], $case[1]),
+            ];
+        }
+
+        yield 'encapsed followed by simple double quoted 1' => [
+            '<?php echo "Hello, {$fruit}s.";',
+            '<?php echo \'Hello,\'  .  " {$fruit}s.";',
+        ];
+
+        yield 'encapsed followed by simple double quoted 2' => [
+            '<?php echo "Hello.He drank some juice made of {$fruit}s.Bye $user!" /*1*//*2*/ /*3*//*4*/;',
+            '<?php echo \'Hello.\' /*1*/ . /*2*/ "He drank some juice made of {$fruit}s."/*3*/./*4*/"Bye $user!";',
+        ];
+
+        yield [
+            '<?php
+$string = "foobar";
+echo "char @ -4 \"[$string[-2]]\"!";
+',
+            '<?php
+$string = "foobar";
+echo "char @ -4 \"[$string[-2]"."]\"!";
+',
+        ];
+
+        yield 'double quote concat double quote + comment' => [
+            '<?php $fi = "lk" /* 1 *//* 2 */ ;',
+            '<?php $fi = "l" /* 1 */ . /* 2 */ "k";',
+        ];
+
+        yield 'empty concat empty' => [
+            '<?php $aT = "";',
+            '<?php $aT = ""."";',
+        ];
+
+        yield 'multiple fixes' => [
+            '<?php $f0 = "abc | defg";',
+            '<?php $f0 = "a"."b"."c | "."d"."e"."f"."g";',
+        ];
+
+        yield 'linebreak with indent inside' => [
+            '<?php
+$text1 = "intro:   |   |"."
+               line 2 indent";',
+            '<?php
+$text1 = "intro:   |"."   |"."
+               line 2 indent"."";',
+        ];
+
+        yield 'linebreak with indent inside + comment' => [
+            '<?php
+$text2 = "intro:      "." #a
+               line 2 indent";',
+            '<?php
+$text2 = "intro:   "."   "." #a
+               line 2 indent"."";',
+        ];
+
+        yield 'do not fix' => [
```

Comment:
Please add cases such as
* `"$a".'b'`
* `"$a->b".'c'`

Also I think it would be great to have those as separate cases. Having smaller cases eases debugging.

Reply:
These cases are as separate cases in the `$cases` array.